src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -429,28 +429,22 @@ public virtual HashAlgorithm GetHashAlgorithm(string algorithm)
             {
                 switch (algorithm)
                 {
-                    case SecurityAlgorithms.SHA256:
-                    case SecurityAlgorithms.ECDSA_SHA256:
-                    case SecurityAlgorithms.HMAC_SHA256:
-                    case SecurityAlgorithms.RSA_SHA256:
-                    case SecurityAlgorithms.RsaSha256Signature:
-                    case SecurityAlgorithms.PS256:
+                    case JwtAlgorithms.ECDSA_SHA256:
```

Comment:
You are turning this into: JwtAsymmetricSignatureProvider. It is general purpose and needs to serve other algorithms.